File: go-sdk/pkg/models/operations/postshieldinformationbarriers.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PostShieldInformationBarriersRequest struct {
	Retries *utils.RetryConfig
	Request *shared.ShieldInformationBarrier `request:"mediaType=application/json"`
}

type PostShieldInformationBarriersResponse struct {
	ClientError              *shared.ClientError
	ContentType              string
	ShieldInformationBarrier *shared.ShieldInformationBarrier
	StatusCode               int64
}

File: ts-sdk/src/sdk/models/operations/postshieldinformationbarriers.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostShieldInformationBarriersRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.ShieldInformationBarrier;
}


export class PostShieldInformationBarriersResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  shieldInformationBarrier?: shared.ShieldInformationBarrier;

  @SpeakeasyMetadata()
  statusCode: number;
}
